src/optim/FishLeg/fishleg.py

```diff
@@ -278,34 +283,99 @@ def update_aux(self) -> None:
                 u2 = u2 + torch.sum(params.grad * params.grad)
         u_norm = torch.sqrt(u2)
 
-        reg_term = 0.0
-        quad_term = 0.0
-        linear_term = 0.0
+        aux_loss = 0
 
-        p_idx = 0
-        for module in self.model.modules():
-            if not isinstance(module, nn.Sequential) and hasattr(module, "Qv"):
-                layer_grads = []
-                for name, param in module.named_parameters():
-                    if "fishleg_aux" not in name:
-                        layer_grads.append(param.grad.data / u_norm)
+        if self.method == "antithetic":
+            nat_grads = []
+            p_idx = 0
+            for module in self.model.modules():
+                if not isinstance(module, nn.Sequential) and hasattr(module, "Qv"):
+                    layer_grads = []
+                    for name, param in module.named_parameters():
+                        if "fishleg_aux" not in name:
+                            layer_grads.append(param.grad.data)
+
+                    qv, qv_b = module.Qv(layer_grads)
+                    aux_loss -= torch.sum(grads[p_idx] * qv)
+                    aux_loss -= torch.sum(grads[p_idx + 1] * qv_b)
+
+                    aux_loss += damping / 2 * torch.sum(qv**2)
+                    aux_loss += damping / 2 * torch.sum(qv_b**2)
+
+                    nat_grads.append(qv)
+                    nat_grads.append(qv_b)
+                    p_idx += 2
```

Comment:
solution compatible with other layer types:

layer_nat_grads = module.Qv(layer_grads)
for i, gr in enumerate(layer_nat_grads):
    aux_loss -= torch.sum(grads[p_idx + i] * gr)
    aux_loss += damping / 2 * torch.sum(gr**2)
    nat_grads.append(gr)
p_idx += i